tests/core/test_dogstatsd.py

```diff
@@ -32,18 +37,29 @@ def test_get_socket_address(self):
 
     @mock.patch('dogstatsd.get_config')
     @mock.patch('dogstatsd.Server')
-    def test_init(self, s, gc):
+    def test_init5(self, s, gc):
         gc.return_value = defaultdict(str)
         gc.return_value['non_local_traffic'] = True
         gc.return_value['use_dogstatsd'] = True
 
-        init()
+        init5()
 
         # if non_local_traffic was passed, use IPv4 wildcard
         s.assert_called_once()
         args, _ = s.call_args
         self.assertEqual(args[1], '0.0.0.0')
 
+    @mock.patch('dogstatsd.get_config')
+    def test_init6(self, gc):
+        cfg = defaultdict(str)
+        cfg['api_key'] = "deadbeeffeebdaed"
+        cfg['use_dogstatsd'] = True
+        cfg['dogstatsd6_enable'] = True
+        gc.return_value = cfg
+
+        path, _, env = init6(gc)
+        self.assertNotEqual(env, os.environ)
+        self.assertEqual(env.get('DD_API_KEY'), cfg['api_key'])
```

Comment:
also, I think we should configure `dogstatsd6` so that it logs to a `dogstatsd6.log` file in the same directory as the one that's configured for `dogstatsd` (if it's the same file I could see some weird stuff happening with the log rotation of `dogstatsd.py` vs `dogstatsd6`... May be completely fine to do that though)